.github/workflows/ci.yml

```diff
@@ -0,0 +1,144 @@
+        if: contains(matrix.os, 'ubuntu') && matrix.build.cc == 'clang-10'
+        run: |
+          sudo -E apt-get install --fix-missing -qq -o Acquire::Retries=3 \
+            clang-10 libomp-10-dev lld-10 llvm-10
+
+      - name: Prepare build environment
+        run: |
+          echo "CC=${{ matrix.build.cc }}" >> $GITHUB_ENV
+          echo "CXX=${{ matrix.build.cxx }}" >> $GITHUB_ENV
+          echo "LD=${{ matrix.build.linker }}" >> $GITHUB_ENV
+          echo "CPPFLAGS=-Wall" >> $GITHUB_ENV
+
+      - name: Prepare Ubuntu environment
+        if: contains(matrix.os, 'ubuntu')
+        run: echo "JOBS=$(nproc)" >> $GITHUB_ENV
+
+      - name: Prepare macOS environment
+        if: contains(matrix.os, 'macos')
+        run: |
+          echo "JOBS=$(sysctl -n hw.logicalcpu)" >> $GITHUB_ENV
+          echo "PKG_CONFIG_PATH=/usr/local/opt/jpeg-turbo/lib/pkgconfig:/usr/local/opt/libxml2/lib/pkgconfig" >> $GITHUB_ENV
+
+      - name: Prepare sanitizers
+        if: matrix.sanitize
+        env:
+          LLVM_PREFIX: /usr/lib/llvm-10
+        run: |
+          ASAN_DSO=`$CC -print-file-name=libclang_rt.asan-x86_64.so`
+          echo "LDSHARED=$CC -shared" >> $GITHUB_ENV
+          echo "CPPFLAGS=-g -fsanitize=address,undefined -fno-omit-frame-pointer -fopenmp -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION" >> $GITHUB_ENV
+          echo "LDFLAGS=-g -fsanitize=address,undefined -shared-libasan -fopenmp=libomp" >> $GITHUB_ENV
+          echo "ASAN_DSO=$ASAN_DSO" >> $GITHUB_ENV
+          echo "ASAN_OPTIONS=suppressions=${{ github.workspace }}/suppressions/asan.supp" >> $GITHUB_ENV
+          echo "LSAN_OPTIONS=suppressions=${{ github.workspace }}/suppressions/lsan.supp" >> $GITHUB_ENV
+          echo "UBSAN_OPTIONS=suppressions=${{ github.workspace }}/suppressions/ubsan.supp:print_stacktrace=1" >> $GITHUB_ENV
+          echo "LD_LIBRARY_PATH=$LLVM_PREFIX/lib:`dirname $ASAN_DSO`" >> $GITHUB_ENV
+          echo "$LLVM_PREFIX/bin" >> $GITHUB_PATH
+          # workaround for https://github.com/google/sanitizers/issues/89
+          # otherwise libIlmImf-2_3.so ends up as <unknown module>
+          echo "DLCLOSE_PRELOAD=${{ github.workspace }}/dlclose.so" >> $GITHUB_ENV
+          echo -e '#include <stdio.h>\nint dlclose(void*handle){return 0;}' | $CC -shared -xc -odlclose.so -
+
+      - name: Configure libvips
+        run: |
+          ./autogen.sh \
+            --disable-dependency-tracking \
+            --disable-deprecated || (cat config.log && exit 1)
+
+      - name: Build libvips
+        run: make V=0 -j$JOBS
+
+      - name: Check libvips
+        run: make V=0 check
+
+      - name: Install libvips
+        run: sudo make V=0 install
+
+      - name: Rebuild the shared library cache
+        if: contains(matrix.os, 'ubuntu')
+        run: sudo ldconfig
+
+      - name: Install pyvips
+        run: pip3 install pyvips[test]
+
+      - name: Run test suite
+        env:
+          VIPS_LEAK: 1
```

Comment:
Oh, interesting, does `VIPS_LEAK=1 python ...` not work? Or is this just neater?

Reply:
That works too, but I thought I'd add it here, otherwise the length of the command gets quite long (if `LD_PRELOAD` is also included).

Reply:
Ah gotcha. Yes, it's neater, I agree.